src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.a < 0 || this.b < 0 || this.c < 0) {
+      throw new Error('Invalid side');
+    }
 
-export class Circle {
+    if (a >= b + c || b >= a + c || c >= b + a) {
+      throw new Error('This is not triangle');
```

Comment:
3 point in checklist

Reply:
agree, make your error messages more descriptive